x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/formula/validation.ts

```diff
@@ -224,6 +284,12 @@ export function tryToParse(
   try {
     root = parse(formula);
   } catch (e) {
+    // give it last try
+    const maybeQueryProblems = getRawQueryValidationError(formula);
+    if (maybeQueryProblems) {
+      // need to emulate an error shape here
+      return { root: null, error: { message: maybeQueryProblems[0], locations: [] } };
+    }
```

Comment:
This section of code confused me at first, reading through it looks like all I need is unparseable code and any string that matches `kql=` or `lucene=`. This could lead to false positives, for example if the entire text is `kql=` it'll trigger here. Overall I do think this case is helpful to users, but I'd like you to add some code comments and test cases for these false positives.